### PR TITLE
Add various tree-sitter languages

### PR DESCRIPTION
**Pull request by VixieTSQ**
Partially fixes https://github.com/lapce/lapce/issues/272. 

I haven't tested any of these, and they really should be tested. We could @ some people in the discord server who have the OS testing roles but I doubt a lot of them read that the roles were for testing...

### CODE REVIEW

File: lapce-core/src/language.rs [hunk 1]
```diff
@@ -376,6 +388,66 @@ const LANGUAGES: &[SyntaxProperties] = &[
         code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
         extensions: &["hs"],
     },
+    #[cfg(feature = "lang-glimmer")]
+    SyntaxProperties {
+        id: LapceLanguage::Glimmer,
+        language: tree_sitter_glimmer::language,
+        highlight: tree_sitter_glimmer::HIGHLIGHTS_QUERY,
+        comment: "{{!",
```
Comment (VixieTSQ) [inline lapce-core/src/language.rs:396]: Glimmer comments in reality look like `{{! text here }}` but our current system here doesn't allow for comments like that.

File: lapce-core/src/language.rs [hunk 2]
```diff
@@ -376,6 +388,66 @@ const LANGUAGES: &[SyntaxProperties] = &[
...
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["hcl"],
+    },
+    #[cfg(feature = "lang-ocaml")]
+    SyntaxProperties {
+        id: LapceLanguage::OCaml,
+        language: tree_sitter_ocaml::language_ocaml,
+        highlight: tree_sitter_ocaml::HIGHLIGHTS_QUERY,
+        comment: "(*",
```
Comment (VixieTSQ) [inline lapce-core/src/language.rs:426]: Ocaml comments in reality look like `(* text here *)` but our current system here doesn't allow for comments like that.

File: lapce-core/src/language.rs [hunk 3]
```diff
@@ -376,6 +388,66 @@ const LANGUAGES: &[SyntaxProperties] = &[
...
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["mli"],
+    },
+    #[cfg(feature = "lang-scss")]
+    SyntaxProperties {
+        id: LapceLanguage::SCSS,
+        language: tree_sitter_scss::language,
+        highlight: tree_sitter_scss::HIGHLIGHTS_QUERY,
+        comment: "//",
```
Comment (VixieTSQ) [inline lapce-core/src/language.rs:446]: SCSS also has support for comments like `/* text here */` but our current system doesn't allow for multiple comment types.

**Review by VixieTSQ — COMMENTED** [review]
Comments really need to be reworked for Lapce. We could just let that field take a closure that returns string slices per a given string.

File: lapce-core/src/language.rs (hunk 3 above)
Comment (VixieTSQ) [reply lapce-core/src/language.rs:446]: > Comments really need to be reworked for Lapce. We could just let that field take a closure that returns string slices per a given string.

Now that I think about, this comment field is probably used for automatically commenting things with a shortcut, isn't it? In that case, one comment only makes sense but still isn't preferable honestly, we might wanna have a thing where it uses one comment type for multi line comments and vice versa. But all of this still is a problem with languages that have comments with two symbols on the sides of your text.

**Comment by VixieTSQ** [comment]
> Comments really need to be reworked for Lapce. We could just let that field take a closure that returns string slices per a given string.

Now that I think about, this comment field is probably used for automatically commenting things with a shortcut, isn't it? In that case, one comment only makes sense but still isn't preferable honestly, we might wanna have a thing where it uses one comment type for multi line comments and vice versa. But all of this still is a problem with languages that have comments with two symbols on the sides of your text.

**Comment by dzhou121** [comment]
> > Comments really need to be reworked for Lapce. We could just let that field take a closure that returns string slices per a given string.
> 
> Now that I think about, this comment field is probably used for automatically commenting things with a shortcut, isn't it? In that case, one comment only makes sense but still isn't preferable honestly, we might wanna have a thing where it uses one comment type for multi line comments and vice versa. But all of this still is a problem with languages that have comments with two symbols on the sides of your text.

We could create an issue for supporting multi-line comment.

**Comment by bugadani** [comment]
Please don't mention "fixes #issue" in the OP unless the PR resolves the complete issue.

**Comment by VixieTSQ** [comment]
> Please don't mention "fixes #issue" in the OP unless the PR resolves the complete issue.

I didn't know github does that, sorry!

**Pull request merged**